src/cc2olx/iframe_link_parser.py

```diff
@@ -0,0 +1,167 @@
+from urllib.parse import parse_qs, urlparse
+
+from cc2olx.link_file_reader import LinkFileReader
+from cc2olx.utils import element_builder
+
+
+class IframeLinkParser:
+    """
+    This class forms the base class for all type of link extractor that are being
+    written.
+    """
+
+    def __init__(self, link_file):
+        self.link_map = LinkFileReader(link_file).get_link_map()
+
+    def _extract_src(self, iframe_element):
+        """
+        Extract the link that iframe is embedding inside it.
+
+        Args:
+            iframe_element ([Xml Element]): The iframe tag that is found.
+
+        Returns:
+            [str]: The whole URL string which was embedded inside iframe.
+        """
+        return iframe_element.attrib['src']
+
+    def _get_video_url(self, iframes):
+        """
+        This function is responsible to form a proper video URL from the
+        polluted URL embeded inside the iframe.
+
+        Args:
+            iframes ([Xml Element]): Single iframe tag that is extracted.
+
+        Returns:
+            List[str]: List of urls for the iframes.
+        """
+        urls = []
+        for iframe in iframes:
+            src = self._extract_src(iframe)
+            url = self._extract_url(src)
+            urls.append(url)
+        return urls
+
+    def _extract_url(self, url):
+        """
+        This function helps to extract the URL, this URL is the one formed to extract
+        the URL that is present in the CSV file. Since the URL has to be formed because
+        of the distortion in the iframe embedded.
+
+        Args:
+            url ([str]): The iframe embedded URL
+        """
+        raise NotImplementedError
+
+    def get_video_olx(self, doc, iframes):
+        """
+        The public function that helps to generate and collect all the
+        video OLX and sends it to get written.
+
+        Args:
+            doc (XML Document): The document on which the child is formed
+            iframes (List[XMl iframe]): List of all the iframes found on the page
+
+        Returns:
+            List[Video OLX]: List of video OLX children that is being formed.
+        """
+        video_olx_list = []
+        video_urls = self._get_video_url(iframes)
+        for url, iframe in zip(video_urls, iframes):
+            url_row = self.link_map.get(url)
+            if url_row:
+                video_olx = self._create_video_olx(doc, url_row)
+                video_olx_list.append(video_olx)
+                # Remove the iframe from parent since it's already an xblock now.
+                iframe.getparent().remove(iframe)
```

Comment:
Why remove the iframe?

Reply:
if we don't then we will have iframe with video and also a video block, which was AFAIK no required, what do you think?